datafusion/common/src/hash_utils.rs

```diff
@@ -267,6 +268,38 @@ where
     Ok(())
 }
 
+fn hash_fixed_list_array(
```

Comment:
I didn't see any test coverage for this new code -- e.g. either unit tests for hashing or a higher level test like `GROUP BY <FixedListArray>`

Can you either ensure this code is tested somehow, or else perhaps move the hash support to a different PR so we can merge the `arrow_cast` support ?